src/tracer.js

```diff
@@ -364,6 +370,10 @@ function sendCurrentTrace(traceSender) {
 
     const sendResult = traceSender(traceJson);
     tracerObj.pendingEvents.clear();
+
+    if (tracerObj.sampleRate !== consts.DEFAULT_SAMPLE_RATE) {
+        tracerObj.deleted = true;
+    }
```

Comment:
you can just check the config here instead of doing it on a per-trace basis (still set deleted on the current trace)

Reply:
Had this thought. for somereason chose to do this per-trace basis.
Looks ok with config. :)
changing!